sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java

```diff
@@ -185,56 +202,44 @@ public Object[][] connectionStateListenerArgProvider() {
-            assertThat(diagnostics).contains("\"serializationType\":\"PARTITION_KEY_FETCH_SERIALIZATION\"");
-            assertThat(diagnostics).contains("\"userAgent\":\"" + Utils.getUserAgent() + "\"");
-            assertThat(diagnostics).containsAnyOf(
-                "\"machineId\":\"" + tempMachineId + "\"", // logged machineId should be static uuid or
-                "\"machineId\":\"" + ClientTelemetry.getMachineId(null) + "\"" // the vmId from Azure
-            );
-            assertThat(diagnostics).containsPattern("(?s).*?\"activityId\":\"[^\\s\"]+\".*");
-            assertThat(createResponse.getDiagnostics().getDuration()).isNotNull();
-            assertThat(createResponse.getDiagnostics().getContactedRegionNames()).isNotNull();
-            // TODO: (nakumars) - Uncomment the following line after your client telemetry fix
-            // assertThat(createResponse.getDiagnostics().getRegionsContacted()).isNotEmpty();
-            validateTransportRequestTimelineGateway(diagnostics);
-            validateRegionContacted(createResponse.getDiagnostics(), testGatewayClient.asyncClient());
-            isValidJSON(diagnostics);
-        } finally {
-            if (testGatewayClient != null) {
-                testGatewayClient.close();
-            }
-        }
+
+        // Adding a delay to allow async VM instance metadata initialization to complete
+        Thread.sleep(2000);
+        InternalObjectNode internalObjectNode = getInternalObjectNode();
+        CosmosItemResponse<InternalObjectNode> createResponse = containerGateway.createItem(internalObjectNode);
+        String diagnostics = createResponse.getDiagnostics().toString();
+
+        assertThat(diagnostics).contains("\"connectionMode\":\"GATEWAY\"");
+        assertThat(diagnostics).contains("\"userAgent\":\"" + this.gatewayClientUserAgent + "\"");
+        assertThat(diagnostics).doesNotContain(("\"gatewayStatistics\":null"));
+        assertThat(diagnostics).contains("\"operationType\":\"Create\"");
+        assertThat(diagnostics).contains("\"metaDataName\":\"CONTAINER_LOOK_UP\"");
+        assertThat(diagnostics).contains("\"serializationType\":\"PARTITION_KEY_FETCH_SERIALIZATION\"");
+        assertThat(diagnostics).contains("\"userAgent\":\"" + this.gatewayClientUserAgent + "\"");
+        assertThat(diagnostics).containsAnyOf(
+            "\"machineId\":\"" + tempMachineId + "\"", // logged machineId should be static uuid or
+            "\"machineId\":\"" + ClientTelemetry.getMachineId(null) + "\"" // the vmId from Azure
+        );
+        assertThat(diagnostics).containsPattern("(?s).*?\"activityId\":\"[^\\s\"]+\".*");
+        assertThat(createResponse.getDiagnostics().getDuration()).isNotNull();
+        assertThat(createResponse.getDiagnostics().getContactedRegionNames()).isNotNull();
+        // TODO: (nakumars) - Uncomment the following line after your client telemetry fix
```

Comment:
uncomment or delete?

Reply:
Uncommented the assertions and removed the comments